src/tour/treasury.md

```diff
@@ -1,19 +1,92 @@
-# srml-treasury
+# smpl-treasury
 
-* keeps the proportion of tokens staked constant
-* why? security in PoS systems
-* see `OnDilution` trait as well as the facts regarding dilution (where can I show in other files where dilution is taking place)
+This recipe demonstrates how [srml/treasury](https://github.com/paritytech/substrate/blob/master/srml/treasury/src/lib.rs) instantiates a pot of funds and schedules funding. *See [kitchen/treasury](https://github.com/substrate-developer-hub/recipes/tree/master/kitchen/treasury) for the full code*
 
-* the `Imbalances` type provides safe-esque accounting
+## Instantiate a Pot
 
-* discuss how proposals are collateralized with `reserve` and `unreserve` `=>` sybil mechanism basically
+To instantiate a pool of funds, import [`ModuleId`](https://crates.parity.io/sr_primitives/struct.ModuleId.html) and [`AccountIdConversion`](https://crates.parity.io/sr_primitives/traits/trait.AccountIdConversion.html) from [sr-primitives](https://crates.parity.io/sr_primitives/index.html).
 
-* scheduling spending funds with `on_finalize` `=>` this could supplement `blockchain event loop`...keep the recipe in `balances` for schedulinhg execution as a minimal example (could provide commentary with it in the repo instead of in the book...?)...use it to introduce treasury logic for scheduling execution quickly `=>` then reference the other recipe
+```rust
+use runtime_primitives::{ModuleId, traits::AccountIdConversion};
+```
 
-## open questions
+With these imports, a `MODULE_ID` constant can be generated as an identifier for the pool of funds. This identifier can be converted into an `AccountId` with the `into_account()` method provided by the [`AccountIdConversion`](https://crates.parity.io/sr_primitives/traits/trait.AccountIdConversion.html) trait.
 
-If we read [here](), we realize that there are a few things that are not yet done for this treasury module. How do assess proposals for spending? 
+```rust
+const MODULE_ID: ModuleId = ModuleId(*b"example ");
 
-* set spending targets
-* set inflow targets based on total stake amount (targets basically)
-* use futarchy for governance of changes and price proposals in proportion to how far they veer from norms `=>` allow crowdfunding proposals with proportional governance?
+impl<T: Trait> Module<T> {
+    pub fn account_id() -> T::AccountId {
+		MODULE_ID.into_account()
+	}
+
+    fn pot() -> BalanceOf<T> {
+		T::Currency::free_balance(&Self::account_id())
+	}
+}
+```
+
+Accessing the pot's balance is as simple as using the [`Currency`](https://crates.parity.io/srml_support/traits/trait.Currency.html) trait to access the balance of the associated `AccountId`.
+
+## Proxy Transfers
```

Comment:
In a vacuum, this section did not make so much sense to me.

* Why are we using a vec?
* What pattern should I learn here?

Reply:
I'll add some context and make this more clear. As I mentioned to you before, it's just filling a dispatch queue and then it blackboxes how in `spend_funds` the `Vec` is iterated through the dispatch queue and the funds are spent until the pot is empty or the queue is empty. This is scheduled as detailed in the next small section. So the narrative which I have to communicate more clearly is that the treasury contains these three useful patterns:
* instantiate a pot
* use a dispatch queue for approved spend requests
* schedule batched execution of spend requests

I also remember you mentioning that list iteration in the runtime is evil and I responded without really thinking and said I could do the iteration in off-chain workers. That doesn't make sense in this context because either way I'm going to have to execute all these spend requests in the runtime. The cost of executing spend requests scales linearly with spend requests -- the way to manage this cost is to set limits on the number of approved spend requests by maybe setting a limit on the number of entries approved every `SpendPeriod`. This is definitely worth mentioning though and I'm glad you brought it up...will add this soon!